datafusion/sqllogictest/test_files/predicates.slt

```diff
@@ -725,3 +725,57 @@ AggregateExec: mode=SinglePartitioned, gby=[p_partkey@2 as p_partkey], aggr=[SUM
 --------CoalesceBatchesExec: target_batch_size=8192
 ----------RepartitionExec: partitioning=Hash([ps_partkey@0], 4), input_partitions=1
 ------------MemoryExec: partitions=1, partition_sizes=[1]
+
+# Inlist simplification
+
+statement ok
+create table t(x int) as values (1), (2), (3);
+
+query TT
+explain select x from t where x IN (1,2,3) AND x IN (4,5);
```

Comment:
i have some additional suggested tests:
1. Negative tests like `x IN (1,2,3) OR x IN (4,5)` (maybe that could actually be combined into `X IN (1,2,3,4,5)` 🤔 
2. Tests with more than 2 clauses: `x IN (1,2,3) AND x IN (3,4) AND x IN (2,3))`
3. Tests with more than 2 clauses that are not all IN lists but the inlists could be combined `x IN (1,2,3) AND y = 2 AND x IN (2,3))` 

Maybe also some tests that could in theory be simplified like


```
`x IN (1,2,3) AND x = 2 AND x IN (2,3))`   --> x=2
```

(we don't have to actually implement that simplification in this PR, but it would be good to document the potential further improvement in tests)